test_numpy_type fails under Python 3.8

When building 2.4 on an openSUSE system with Python 3.8, I get the following failure. I'm at a loss figuring out why some of the attributes are outputted in a different order.

```
[  242s] =================================== FAILURES ===================================
[  242s] ___________________________ TestGEXF.test_numpy_type ___________________________
[  242s] 
[  242s] self = <networkx.readwrite.tests.test_gexf.TestGEXF object at 0x7fe159d69310>
[  242s] 
[  242s]         def test_numpy_type(self):
[  242s]             G = nx.path_graph(4)
[  242s]             try:
[  242s]                 import numpy
[  242s]             except ImportError:
[  242s]                 return
[  242s]             nx.set_node_attributes(G, {n: n for n in numpy.arange(4)}, 'number')
[  242s]             G[0][1]['edge-number'] = numpy.float64(1.1)
[  242s]     
[  242s]             expected = """<gexf version="1.2" xmlns="http://www.gexf.net/1.2draft"\
[  242s]      xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation\
[  242s]     ="http://www.gexf.net/1.2draft http://www.gexf.net/1.2draft/gexf.xsd">
[  242s]       <meta lastmodifieddate="{}">
[  242s]         <creator>NetworkX {}</creator>
[  242s]       </meta>
[  242s]       <graph defaultedgetype="undirected" mode="static" name="">
[  242s]         <attributes class="edge" mode="static">
[  242s]           <attribute id="1" title="edge-number" type="float" />
[  242s]         </attributes>
[  242s]         <attributes class="node" mode="static">
[  242s]           <attribute id="0" title="number" type="int" />
[  242s]         </attributes>
[  242s]         <nodes>
[  242s]           <node id="0" label="0">
[  242s]             <attvalues>
[  242s]               <attvalue for="0" value="0" />
[  242s]             </attvalues>
[  242s]           </node>
[  242s]           <node id="1" label="1">
[  242s]             <attvalues>
[  242s]               <attvalue for="0" value="1" />
[  242s]             </attvalues>
[  242s]           </node>
[  242s]           <node id="2" label="2">
[  242s]             <attvalues>
[  242s]               <attvalue for="0" value="2" />
[  242s]             </attvalues>
[  242s]           </node>
[  242s]           <node id="3" label="3">
[  242s]             <attvalues>
[  242s]               <attvalue for="0" value="3" />
[  242s]             </attvalues>
[  242s]           </node>
[  242s]         </nodes>
[  242s]         <edges>
[  242s]           <edge id="0" source="0" target="1">
[  242s]             <attvalues>
[  242s]               <attvalue for="1" value="1.1" />
[  242s]             </attvalues>
[  242s]           </edge>
[  242s]           <edge id="1" source="1" target="2" />
[  242s]           <edge id="2" source="2" target="3" />
[  242s]         </edges>
[  242s]       </graph>
[  242s]     </gexf>""".format(time.strftime('%Y-%m-%d'), nx.__version__)
[  242s]             obtained = '\n'.join(nx.generate_gexf(G))
[  242s] >           assert expected == obtained
[  242s] E           assert '<gexf versio...aph>\n</gexf>' == '<gexf xmlns="...aph>\n</gexf>'
[  242s] E             - <gexf version="1.2" xmlns="http://www.gexf.net/1.2draft" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.gexf.net/1.2draft http://www.gexf.net/1.2draft/gexf.xsd">
[  242s] E             ?       --------------
[  242s] E             + <gexf xmlns="http://www.gexf.net/1.2draft" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.gexf.net/1.2draft http://www.gexf.net/1.2draft/gexf.xsd" version="1.2">
[  242s] E             ?                                                                                                                                                                                        ++++++++++++++
[  242s] E                 <meta lastmodifieddate="2019-11-13">
[  242s] E                   <creator>NetworkX 2.4</creator>
[  242s] E                 </meta>
[  242s] E                 <graph defaultedgetype="undirected" mode="static" name="">
[  242s] E             -     <attributes class="edge" mode="static">
[  242s] E             +     <attributes mode="static" class="edge">
[  242s] E                     <attribute id="1" title="edge-number" type="float" />
[  242s] E                   </attributes>
[  242s] E             -     <attributes class="node" mode="static">
[  242s] E             +     <attributes mode="static" class="node">
[  242s] E                     <attribute id="0" title="number" type="int" />
[  242s] E                   </attributes>
[  242s] E                   <nodes>
[  242s] E                     <node id="0" label="0">
[  242s] E                       <attvalues>
[  242s] E                         <attvalue for="0" value="0" />
[  242s] E                       </attvalues>
[  242s] E                     </node>
[  242s] E                     <node id="1" label="1">
[  242s] E                       <attvalues>
[  242s] E                         <attvalue for="0" value="1" />
[  242s] E                       </attvalues>
[  242s] E                     </node>
[  242s] E                     <node id="2" label="2">
[  242s] E                       <attvalues>
[  242s] E                         <attvalue for="0" value="2" />
[  242s] E                       </attvalues>
[  242s] E                     </node>
[  242s] E                     <node id="3" label="3">
[  242s] E                       <attvalues>
[  242s] E                         <attvalue for="0" value="3" />
[  242s] E                       </attvalues>
[  242s] E                     </node>
[  242s] E                   </nodes>
[  242s] E                   <edges>
[  242s] E             -       <edge id="0" source="0" target="1">
[  242s] E             ?            -------
[  242s] E             +       <edge source="0" target="1" id="0">
[  242s] E             ?                                  +++++++
[  242s] E                       <attvalues>
[  242s] E                         <attvalue for="1" value="1.1" />
[  242s] E                       </attvalues>
[  242s] E                     </edge>
[  242s] E             -       <edge id="1" source="1" target="2" />
[  242s] E             ?            -------
[  242s] E             +       <edge source="1" target="2" id="1" />
[  242s] E             ?                                   +++++++
[  242s] E             -       <edge id="2" source="2" target="3" />
[  242s] E             ?            -------
[  242s] E             +       <edge source="2" target="3" id="2" />
[  242s] E             ?                                   +++++++
[  242s] E                   </edges>
[  242s] E                 </graph>
[  242s] E               </gexf>
[  242s] 
[  242s] networkx/readwrite/tests/test_gexf.py:459: AssertionError
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -73,8 +73,6 @@ matrix:
           - libgdal-dev
           - graphviz
     - python: pypy3.6-7.1.1
-  allow_failures:
-    - python: 3.8
 
 before_install:
   # prepare the system to install prerequisites or dependencies

--- a/networkx/readwrite/tests/test_gexf.py
+++ b/networkx/readwrite/tests/test_gexf.py
@@ -404,7 +404,8 @@ gexf.net/1.2draft http://www.gexf.net/1.2draft/gexf.xsd" version="1.2">
         nx.set_node_attributes(G, {n: n for n in numpy.arange(4)}, 'number')
         G[0][1]['edge-number'] = numpy.float64(1.1)
 
-        expected = """<gexf version="1.2" xmlns="http://www.gexf.net/1.2draft"\
+        if sys.version_info < (3, 8):
+            expected = """<gexf version="1.2" xmlns="http://www.gexf.net/1.2draft"\
  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation\
 ="http://www.gexf.net/1.2draft http://www.gexf.net/1.2draft/gexf.xsd">
   <meta lastmodifieddate="{}">
@@ -447,6 +448,54 @@ gexf.net/1.2draft http://www.gexf.net/1.2draft/gexf.xsd" version="1.2">
       </edge>
       <edge id="1" source="1" target="2" />
       <edge id="2" source="2" target="3" />
+    </edges>
+  </graph>
+</gexf>""".format(time.strftime('%Y-%m-%d'), nx.__version__)
+        else:
+            expected = """<gexf xmlns="http://www.gexf.net/1.2draft"\
+ xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation\
+="http://www.gexf.net/1.2draft http://www.gexf.net/1.2draft/gexf.xsd"\
+ version="1.2">
+  <meta lastmodifieddate="{}">
+    <creator>NetworkX {}</creator>
+  </meta>
+  <graph defaultedgetype="undirected" mode="static" name="">
+    <attributes mode="static" class="edge">
+      <attribute id="1" title="edge-number" type="float" />
+    </attributes>
+    <attributes mode="static" class="node">
+      <attribute id="0" title="number" type="int" />
+    </attributes>
+    <nodes>
+      <node id="0" label="0">
+        <attvalues>
+          <attvalue for="0" value="0" />
+        </attvalues>
+      </node>
+      <node id="1" label="1">
+        <attvalues>
+          <attvalue for="0" value="1" />
+        </attvalues>
+      </node>
+      <node id="2" label="2">
+        <attvalues>
+          <attvalue for="0" value="2" />
+        </attvalues>
+      </node>
+      <node id="3" label="3">
+        <attvalues>
+          <attvalue for="0" value="3" />
+        </attvalues>
+      </node>
+    </nodes>
+    <edges>
+      <edge source="0" target="1" id="0">
+        <attvalues>
+          <attvalue for="1" value="1.1" />
+        </attvalues>
+      </edge>
+      <edge source="1" target="2" id="1" />
+      <edge source="2" target="3" id="2" />
     </edges>
   </graph>
 </gexf>""".format(time.strftime('%Y-%m-%d'), nx.__version__)

